src/components/App.tsx

```diff
@@ -131,31 +135,46 @@ class App extends Component<Props> {
   render() {
     return (
       <MainContainer>
-        {this.props.view === 'browser' && (
-          <Browser
-            onRunQuery={() => this.props.runQuery({ query: this.props.query })}
-            onSelectQuery={(queryName) => {
-              this.props.resetQueryResults();
-              this.props.selectSavedQuery(queryName, this.props.autorunQuery);
-            }}
-            onEditQuery={(queryName) => {
-              this.props.editQuery(queryName);
-            }}
+        <Switch>
+          <Route
+            exact
+            path={ROUTES.BROWSER}
+            render={() => (
+              <Browser
+                onRunQuery={() =>
+                  this.props.runQuery({ query: this.props.query })
+                }
+                onSelectQuery={(queryName) => {
+                  this.props.resetQueryResults();
+                  this.props.selectSavedQuery(
+                    queryName,
+                    this.props.autorunQuery
+                  );
+                  this.props.push('/browser?savedQuery=' + queryName);
```

Comment:
Instead of inline route definition with `browser` - could we use here `ROUTES` constants and inline savedQuery param.